keras_rcnn/models/_rcnn.py

```diff
@@ -12,78 +12,90 @@
-        output_features = keras.layers.Conv2D(256, name='block3_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(256, name='block3_conv3', **options)(output_features)
-        output_features = keras.layers.Conv2D(256, name='block3_conv4', **options)(output_features)
+        convolution_3x3 = keras.layers.Conv2D(64, **options)(output_features)
 
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block3_pool')(output_features)
+        output_deltas = keras.layers.Conv2D(9 * 4, (1, 1), activation="linear", kernel_initializer="zero", name="deltas")(convolution_3x3)
 
-        output_features = keras.layers.Conv2D(512, name='block4_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv3', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block4_conv4', **options)(output_features)
+        output_scores = keras.layers.Conv2D(9 * 1, (1, 1), activation="sigmoid", kernel_initializer="uniform", name="scores")(convolution_3x3)
 
-        output_features = keras.layers.MaxPooling2D(strides=(2, 2), name='block4_pool')(output_features)
+        target_anchors, target_proposal_categories, target_proposal_bounding_boxes = keras_rcnn.layers.AnchorTarget()([output_scores, target_bounding_boxes, target_metadata])
 
-        output_features = keras.layers.Conv2D(512, name='block5_conv1', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block5_conv2', **options)(output_features)
-        output_features = keras.layers.Conv2D(512, name='block5_conv3', **options)(output_features)
+        output_deltas, output_scores = keras_rcnn.layers.RPN()([target_proposal_bounding_boxes, target_proposal_categories, output_deltas, output_scores])
 
-        convolution_3x3 = keras.layers.Conv2D(512, name="convolution_3x3", **options)(output_features)
+        output_proposal_categories = keras_rcnn.layers.ObjectProposal()([target_metadata, output_deltas, output_scores, target_anchors])
 
-        output_deltas = keras.layers.Conv2D(9 * 4, (1, 1), activation="linear", kernel_initializer="zero", name="deltas")(convolution_3x3)
-        output_scores = keras.layers.Conv2D(9 * 1, (1, 1), activation="sigmoid", kernel_initializer="uniform", name="scores")(convolution_3x3)
+        output_proposal_categories, target_proposal_categories, target_proposal_bounding_boxes = keras_rcnn.layers.ProposalTarget()([output_proposal_categories, target_categories, target_bounding_boxes])
 
-        target_anchors, target_proposal_labels, target_proposals = keras_rcnn.layers.AnchorTarget()([output_scores, target_bounding_boxes, target_metadata])
+        output_features = keras_rcnn.layers.RegionOfInterest((14, 14))([output_features, output_proposal_categories, target_metadata])
```

Comment:
should the 14 be in a config?

Reply:
Good question. 

It’s probably something we wouldn’t encourage users to configure since it’ll usually lead to worse results, but it’s also something users will occasionally need to configure. If we do want to add it, I guess we should make it a parameter of the RCNN model. What do you think?

Reply:
Yeah I'd agree